src/main/java/teammates/common/util/LogHelper.java

```diff
@@ -0,0 +1,192 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import teammates.common.exception.InvalidParametersException;
+
+import com.google.appengine.api.log.AppLogLine;
+import com.google.appengine.api.log.LogQuery;
+import com.google.appengine.api.log.LogServiceFactory;
+import com.google.appengine.api.log.RequestLogs;
+import com.google.appengine.api.log.LogService.LogLevel;
+
+public class LogHelper {
+    /**
+     * 6 versions to query, including the current version and its 5 preceding versions.
+     */
+    private static final int MAX_VERSIONS_TO_QUERY = 6;
+    
+    /**
+     * Always includes application logs
+     */
+    private static final boolean INCLUDE_APP_LOG = true;
+    
+    /**
+     * Affects the internal strategy to get logs. It doesn't affect the result.
+     */
+    private static final int BATCH_SIZE = 1000;
+    private static final LogLevel MIN_LOG_LEVEL = LogLevel.INFO;
+    public static final int SEARCH_TIME_INCREMENT = 2*60*60*1000;  // two hours in millisecond
+    
+    private LogQuery query;
+    private Long endTime;
+    private List<String> versionList;
+    
+    public LogHelper() {
+        query = LogQuery.Builder.withDefaults();
+        query.includeAppLogs(INCLUDE_APP_LOG);
+        query.batchSize(BATCH_SIZE);
+        query.minLogLevel(MIN_LOG_LEVEL);
+    }
```

Comment:
I think you have two classes here: a `AdminLogReader` and `AdminLogQuery`. What do you think?